app/web_ui/src/routes/(app)/settings/manage_tools/[project_id]/add_tools/+page.svelte

```diff
@@ -0,0 +1,176 @@
+          params: {
+            path: {
+              project_id: $page.params.project_id,
+            },
+          },
+          body: {
+            name: name.trim(),
+            server_url: server_url.trim(),
+            headers: headersObj,
+            description: description.trim() || null,
+          },
+        },
+      )
+
+      if (api_error) {
+        throw api_error
+      }
+
+      if (data?.id) {
+        // Navigate to the tools page for the created tool
+        goto(
+          `/settings/manage_tools/${$page.params.project_id}/tools/${data.id}`,
+        )
+      }
+    } catch (e) {
+      error = createKilnError(e)
+    } finally {
+      submitting = false
+    }
+  }
+</script>
+
+<AppPage
+  title="Connect Remote MCP Server"
+  sub_subtitle="Connect to a remote Model Context Protocol (MCP) server to add external
+        tools to your project."
+>
+  <div class="max-w-2xl">
+    <FormContainer
+      submit_label="Connect"
+      on:submit={connect_remote_mcp}
+      bind:error
+      bind:submitting
+    >
+      <FormElement label="Name" id="name" bind:value={name} max_length={120} />
+
+      <FormElement
+        label="Description"
+        inputType="textarea"
+        id="mcp_description"
+        description="A description of this MCP server for your reference."
+        optional={true}
+        bind:value={description}
+      />
+
+      <FormElement
+        label="Server URL"
+        id="mcp_server_url"
+        description="The URL of the remote MCP server"
+        placeholder="https://example.com/mcp"
+        bind:value={server_url}
+      />
+
+      <!-- Headers section -->
+      <FormElement
+        inputType="header_only"
+        label="Headers"
+        id="headers_section"
+        description="If the documentation for the server you're adding requires custom headers, enter them here."
+        optional={true}
+        value=""
+      />
+
+      <FormList
+        content={headers}
+        content_label="Header"
+        start_with_one={false}
+        empty_content={{
+          key: "",
+          value: "",
+        }}
+        let:item_index
+      >
+        <div class="flex gap-2">
+          <div class="flex-1">
+            <FormElement
+              label="Header Name"
+              id="header_name_{item_index}"
+              info_description="The HTTP header name, such as 'Authorization'"
+              placeholder="Header name"
+              light_label={true}
+              bind:value={headers[item_index].key}
+            />
+          </div>
+          <div class="flex-1">
+            <FormElement
+              label="Value"
+              id="header_value_{item_index}"
+              info_description="The header value, such as 'Bearer your-token-here'"
+              placeholder="Value"
+              light_label={true}
+              bind:value={headers[item_index].value}
+            />
+          </div>
+        </div>
+      </FormList>
+    </FormContainer>
```

Comment:
Can we a separate PR but we should think through editing.

If we just want to allow editing name/descripition, we have a `<EditDialog>` control you can add to the tool details page.

If we want to allow editing everyone, we usually make the this entire page into a control, then use on 2 pages, an add_tool page, and and edit_tool page. Edit tool page will fetch existing values and set them via `export name` style vars in the controls